src/content/docs/guides/develop/sidecar.mdx

```diff
@@ -0,0 +1,213 @@
+  }
+  fs.renameSync(
+    `src-tauri/binaries/sidecar${extension}`,
+    `src-tauri/binaries/sidecar-${targetTriple}${extension}`
+  )
+}
+
+main().catch((e) => {
+  throw e
+})
+```
+
+## Running it from Rust
+
+On the Rust side, import the `tauri_plugin_shell::ShellExt` trait and call the `shell().sidecar()` function on the AppHandle:
+
+```rust
+use tauri_plugin_shell::ShellExt;
+use tauri_plugin_shell::process::CommandEvent;
+
+// `sidecar()` expects just the filename, NOT the whole path like in JavaScript
+let sidecar_command = app.shell().sidecar("my-sidecar").unwrap();
+let (mut _rx, mut _child) = sidecar_command
+  .spawn()
+  .expect("Failed to spawn sidecar");
+
+tauri::async_runtime::spawn(async move {
+  // read events such as stdout
+  while let Some(event) = rx.recv().await {
+    if let CommandEvent::Stdout(line) = event {
+      window
+        .emit("message", Some(format!("'{}'", line)))
+        .expect("failed to emit event");
+      // write to stdin
+      child.write("message from Rust\n".as_bytes()).unwrap();
+    }
+  }
+});
+```
+
+You can place this code inside a Tauri command to easily pass the AppHandle or you can store a reference to the AppHandle in the builder script to access it elsewhere in your application.
+
+## Running it from JavaScript
+
+In the JavaScript code, import the `Command` class from the `@tauri-apps/plugin-shell` module and use the `sidecar` static method.
+
+```javascript
+import { Command } from '@tauri-apps/plugin-shell';
+// `binaries/my-sidecar` is the EXACT value specified on `tauri.conf.json > tauri > bundle > externalBin`
+const sidecar_command = Command.sidecar('binaries/my-sidecar')
+const output = await sidecar_command.execute()
+```
+
+## Passing arguments
+
+You can pass arguments to Sidecar commands just like you would for running normal `Command`s.
+
+Arguments can be either **static** (e.g. `-o` or `serve`) or **dynamic** (e.g. `<file_path>` or `localhost:<PORT>`). You define the arguments in the exact order in which you'd call them. Static arguments are defined as-is, while dynamic arguments can be defined using a regular expression.
+
+First, define the arguments that need to be passed to the sidecar command in `src-tauri/capabilities/main.json`:
+
+```json
+{
+  "$schema": "../gen/schemas/desktop-schema.json",
+  "identifier": "default",
+  "description": "Capability for the main window",
+  "windows": ["main"],
+  "permissions": [
+    "path:default",
+    "event:default",
+    "window:default",
+    "app:default",
+    "resources:default",
+    "menu:default",
+    "tray:default",
+    {
+      "identifier": "shell:allow-execute",
+      "allow": [
+        {
+          "args": [
+            "arg1",
+            "-a",
+            "--arg2",
+            {
+              "validator": "\\S+"
+            },
+          ],
+          "cmd": "",
+          "name": "binaries/my-sidecar",
+          "sidecar": true
+        }
+      ]
+    },
+    "shell:allow-open"
+  ]
+}
+```
```

Comment:
Starlight comes with a few niceties like Expressive Code to format code blocks, could you highlight the lines that are to be added/necessary and, if possible, remove unrelated permission
Docs: https://starlight.astro.build/guides/authoring-content/#expressive-code-features
Sample permission codeblock: https://beta.tauri.app/features/stronghold/#permissions

Reply:
for example:
```
```json title="src-tauri/capabilities/main.json" ins={8-14}
```